go-controller/cmd/ovnkube-trace/ovnkube-trace.go

```diff
@@ -898,6 +898,14 @@ func installOvnDetraceDependencies(coreclient *corev1client.CoreV1Client, restco
 		trueFalse := strings.TrimSuffix(depVerifyOut, "\n")
 		klog.V(10).Infof("Dependency check '%s' in pod '%s', container '%s' yielded '%s'", dependencyCmd, podName, "ovnkube-node", trueFalse)
 		if trueFalse != "true" {
+			pipDependencyCmd := "if type -p pip3 >/dev/null 2>&1; then echo 'true' ; fi"
+			pipVerifyOut, pipVerifyErr, err := execInPod(coreclient, restconfig, ovnNamespace, podName, "ovnkube-node", pipDependencyCmd, "")
+			if err != nil {
+				return fmt.Errorf("python3-pip dependency verification check error in pod %s, container %s. Error '%v', stdOut: '%s'\n stdErr: %s",
+					podName, "ovnkube-node", err, pipVerifyOut, pipVerifyErr)
+			}
+			klog.V(10).Infof("python3-pip dependency check '%s' in pod '%s', container '%s' yielded '%s'", pipDependencyCmd,
+				podName, "ovnkube-node", strings.TrimSuffix(pipVerifyOut, "\n"))
```

Comment:
You are too quick or I am too slow!!! Sorry!

I would probably encapsulate this in a function. 

An then change the error below in line 912 to
```
return fmt.Errorf("ovn-detrace error while installing dependency %s in pod %s, container %s. Error '%v', stdOut: '%s'\n stdErr: %s",
    dependency, podName, "ovnkube-node", err, depInstallOut, depInstallErr)
```
So you would use the above error both when the check for pip fails, or when installing the dependency fails.

Reply:
no problem, moved to `verifyDependency` function now, is that ok ?